### DISCUSSION TITLE
Nightscout uploader bug with multiple insulin profiles

### DISCUSSION THREAD

**Discussion by sudousercom** [NightscoutFoundation/xDrip#2089] [General]
Hi! In setting i choosed Fiasp as short, and lantus as long, in xdrip it shows correctly, but prediction line for lantus has 4-6 hours for lantus, but it works all 24 hours.
And then all data uploaded in my nightscout, and it shows basal lantus doses as fast bolus insuline.

How i can fix these issues, or it need to be updated in code?

xdrip version is4f493d3f9-2022.02.17, code 2022021722
nightscout is latest stable version

**Comment by Navid200** [comment]
Please have a look at this:  https://github.com/NightscoutFoundation/xDrip/issues/1158

If you are experiencing the same in Nightscout, please post in the linked issue and state that you are observing the same with Nightscout.

**Reply by sudousercom** [reply]
> If you are experiencing the same in Nightscout, please post in the linked issue and state that you are observing the same with Nightscout.

posted in mentioned issue, thank you

**Comment by sudousercom** [comment]
i have some updates, about fiasp insulin
when i use single insulin setup, fiasp shows activity for 3 hrs, as expected
when multiple insulins enabled, fiasp shows 5 hours curve
